bitbucket/resource_branching_model.go

```diff
@@ -266,7 +266,11 @@ func expandBranchModel(l []interface{}) *BranchModel {
 	rp := &BranchModel{}
 
 	if v, ok := tfMap["name"].(string); ok {
-		rp.Name = v
+		if v == "" {
```

Comment:
I was thinking to maybe use the raw value to actually differentiate between not set vs set as empty value (see https://github.com/hashicorp/terraform-plugin-sdk/issues/741). But I think this would sufice.